exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -425,6 +434,114 @@ static void setResourceForBuilder(
     throw new IllegalArgumentException("Unknown subclass of MonitoredResource.");
   }
 
+  @VisibleForTesting
+  static List<io.opencensus.metrics.export.Metric> summaryMetricToDoubleGaugeMetric(
```

Comment:
I think we need to split the creation of the metric descriptors from the creation of the timeseries. This logic is too complicated and we need to simplify it.